### DISCUSSION TITLE
431 (request headers too large) when user accesses own account

### DISCUSSION THREAD

**Discussion by GregJohnStewart** [keycloak/keycloak#29202] [Q&A] [answered]
Hello, I have the following flow:

1. User logs into my (quarkus based) app fine
2. User navigates to `https://my-keycloak/realms/myrealm/account` via a link on my app
3. Results in blank screen, firefox dev tools say the request is a `431`. No logs in kc output

Interestingly this happened after I started running my keycloak instance with `start` instead of `start-dev`. I don't think it is a realm or permissions issue as such.

I'm imagining it could be from the large number of cookies that are sent with the request?

![image](https://github.com/keycloak/keycloak/assets/7083701/f45a79cc-bdbd-4125-a4ef-15fca9627cd3)

Any ideas?

**Comment by GregJohnStewart** [comment, marked as answer]
Found: https://github.com/keycloak/keycloak/discussions/10236

Looks like `QUARKUS_HTTP_HTTP2=false` is still required in `24.0`. I was able to get it working without `QUARKUS_HTTP_LIMITS_MAX_HEADER_SIZE`

**Reply by codespearhead** [reply]
@GregJohnStewart Thanks for disclosing what worked for you and marking it as the answer, thereby closing the discussion.

While `QUARKUS_HTTP_HTTP2=false` works, it's more of a workaround than an actual solution in my opinion. 

Can you try `QUARKUS_HTTP_LIMITS_MAX_HEADER_SIZE=100k` and tell us in [this answer](https://github.com/keycloak/keycloak/discussions/10236#discussioncomment-4207701) if that works for you? You can comment something along the lines of "I confirm `QUARKUS_HTTP_LIMITS_MAX_HEADER_SIZE=100k` works in Keycloak xx.x.x" and your testimony might help get that discussion closed as well.

Possible duplicates: #29162 , #10236 , #13933 .

**Reply by aazaroual** [reply]
@codespearhead I had the same error in Keycloak 24.0.5 and I solved it by creating /opt/keycloak/24.0.5/conf/quarkus.properties and putting the following lines in it:
```
quarkus.http.limits.max-header-list-size=16384
quarkus.http.limits.max-header-size=100K
```
Restart Keycloak and now it works! **BUT BE CAREFUL: BOTH LINES MUST BE THERE, IF YOU PUT ONLY ONE OF THOSE IT WON'T WORK!**
